core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -4531,6 +4531,23 @@ private void checkNegWindow(String s, String msg) {
         .fails("Table 'SALES.BAD' not found");
   }
 
+    /** Test case for
+     * <a href="https://issues.apache.org/jira/browse/CALCITE-6584">[CALCITE-6584]
+     * Validate prefixed column identifiers in SET clause of UPDATE statement</a>. */
+  @Test void testAliasInSetClauseOfUpdate() {
+    // good examples
+    sql("UPDATE sales.emp AS e SET e.deptno = 10").ok();
+    sql("UPDATE emp AS e SET e.deptno = 10").ok();
```

Comment:
postgres actually rejects this: `Query Error: error: column "e" of relation "emp" does not exist`